File: android/src/main/kotlin/ch/waio/pro_video_editor/src/features/render/helpers/ApplyFlip.kt
import android.util.Log
import androidx.media3.common.Effect
import androidx.media3.common.util.UnstableApi
import androidx.media3.effect.ScaleAndRotateTransformation

@UnstableApi
fun applyFlip(videoEffects: MutableList<Effect>, flipX: Boolean, flipY: Boolean) {
    if (!flipX && !flipY) return

    Log.d(RENDER_TAG, "Applying flip: flipX: $flipX, flipY: $flipY")
    videoEffects += ScaleAndRotateTransformation.Builder()
        .setScale(if (flipX) -1f else 1f, if (flipY) -1f else 1f)
        .build()
}

File: macos/Classes/src/features/render/helpers/ApplyFlip.swift
import CoreGraphics

func applyFlip(
  config: inout VideoCompositorConfig,
  flipX: Bool,
  flipY: Bool
) {
  config.flipX = flipX
  config.flipY = flipY

  if !flipX && !flipY { return }

  print("[\(Tags.render)] Applying flip: flipX=\(flipX), flipY=\(flipY)")
}
